File: Python/Program/GPU.py
from Komponen import Komponen

class GPU(Komponen):
    def __init__(self, memory_size=0, memory_type="", nama="", harga=0, merk=""):
        super().__init__(nama, harga, merk)
        self.memory_size = memory_size
        self.memory_type = memory_type
    
    def set_memory_size(self, memory_size):
        self.memory_size = memory_size
    
    def set_memory_type(self, memory_type):
        self.memory_type = memory_type
    
    def get_memory_size(self):
        return self.memory_size
    
    def get_memory_type(self):
        return self.memory_type
    
    def get_info(self):
        super().get_info()
        print(f"Memory Size: {self.memory_size} GB, Memory Type: {self.memory_type}")

File: Java/Program/GPU.java
public class GPU extends Komponen {
    private int memorySize;
    private String memoryType;

    public GPU() {
    }

    public GPU(int memorySize, String memoryType, String nama, float harga, String merk) {
        super(nama, harga, merk);
        this.memorySize = memorySize;
        this.memoryType = memoryType;
    }

    public void setMemorySize(int memorySize) {
        this.memorySize = memorySize;
    }

    public void setMemoryType(String memoryType) {
        this.memoryType = memoryType;
    }

    public int getMemorySize() {
        return memorySize;
    }

    public String getMemoryType() {
        return memoryType;
    }

    @Override
    public void getInfo() {
        super.getInfo();
        System.out.println("Memory Size: " + memorySize + " GB, Memory Type: " + memoryType);
    }
}